app/scripts/directives/lifecycleHook.js

```diff
@@ -130,4 +130,17 @@ angular.module("openshiftConsole")
         });
       }
     };
+  })
+  .directive("lifecycleHook", function($filter) {
+    return {
+      restrict: 'E',
+      scope: {
+        deploymentConfig: '=',
+        type: '@'  // "pre", "mid", or "post"
+      },
+      templateUrl: 'views/directives/lifecycle-hook.html',
+      controller: function($scope) {
+        $scope.strategyParams = $filter('deploymentStrategyNameToParams')($scope.deploymentConfig.spec.strategy.type);
```

Comment:
You should watch the type in case it changes.

```js
$scope.$watch('deploymentConfig.spec.strategy.type', function(type) {
  $scope.strategyParams = $filter('deploymentStrategyNameToParams')(type);
});
```